my-vault/internal/handlers/handlers_test.go

```diff
@@ -0,0 +1,250 @@
+package handlers
+
+import (
+	"bytes"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/bmasaiti/go-projects/my-vault/internal/domain"
+)
+
+const requestJSON = `{
+  "Id": 34,
+  "name": "server-secret",
+  "kv_map": {
+    "username": "Tamuka",
+    "password": "pass123"
+  }
+}`
+
+const getResponseJSON = `{
+  "message": "Retrieved secret object",
+  "id": "01970123-643a-7d9c-abbf-684a3abe129c",
+  "name": "server-secret",
+  "kv_map": {
+    "password": "pass123",
+    "username": "Tamuka"
+  }
+}`
+
+const responseJSON = `{
+  "message": "Successfully created secret with secretId: 019700cc-d778-7b08-8660-a8965842faf3 and name: server-secret",
+  "id": "019700cc-d778-7b08-8660-a8965842faf3",
+  "name": "server-secret"
+}`
+
+// Mock implementation of SecretsRepository
+type MockSecretsRepository struct {
+	secrets []domain.Secret
+	err     error
+}
```

Comment:
Did you consider rewriting this to using callback function fields as I suggested in #3?